data-node/src/main/java/org/graylog/datanode/opensearch/configuration/beans/impl/SearchableSnapshotsConfigurationBean.java

```diff
@@ -166,17 +180,25 @@ private Map<String, String> properties(boolean searchRoleEnabled) {
         return builder.build();
     }
 
-    private Map<String, String> keystoreItems() {
-        final ImmutableMap.Builder<String, String> builder = ImmutableMap.builder();
+    private Collection<OpensearchKeystoreItem> keystoreItems() {
+        final ImmutableList.Builder<OpensearchKeystoreItem> builder = ImmutableList.builder();
         if (s3RepositoryConfiguration.isRepositoryEnabled()) {
-            builder.put("s3.client.default.access_key", s3RepositoryConfiguration.getS3ClientDefaultAccessKey());
-            builder.put("s3.client.default.secret_key", s3RepositoryConfiguration.getS3ClientDefaultSecretKey());
+            LOG.info("S3 repository configured, adding access and secret key to opensearch keystore");
+            builder.add(new OpensearchKeystoreStringItem("s3.client.default.access_key", s3RepositoryConfiguration.getS3ClientDefaultAccessKey()));
+            builder.add(new OpensearchKeystoreStringItem("s3.client.default.secret_key", s3RepositoryConfiguration.getS3ClientDefaultSecretKey()));
+        }
+
+        if (gcsRepositoryConfiguration.isRepositoryEnabled()) {
+            LOG.info("Google Cloud Storage repository configured, adding credentials file to opensearch keystore");
+            final Path credentialsFile = datanodeDirectories.resolveConfigurationSourceFile(gcsRepositoryConfiguration.getGcsCredentialsFile())
+                    .orElseThrow(() -> new IllegalArgumentException("Failed to resolve Google Cloud Storage credentials file. File not found: " + gcsRepositoryConfiguration.getGcsCredentialsFile()));
+            builder.add(new OpensearchKeystoreFileItem("gcs.client.default.credentials_file", credentialsFile));
         }
         return builder.build();
     }
 
-    private boolean snapshotsAreEnabled() {
-        return s3RepositoryConfiguration.isRepositoryEnabled() || isSharedFileSystemRepo();
+    private boolean snapshotsAreConfigured() {
+        return s3RepositoryConfiguration.isRepositoryEnabled() || isSharedFileSystemRepo() || gcsRepositoryConfiguration.isRepositoryEnabled();
```

Comment:
It would be nice to have a RepositoryConfiguration interface that encapsulates enabled and how to handle the configuration properties. These could be  injected as a set of Repositories int this class. However, as I don't expect any more repo types in the near future, this can be saved for a refactoring some day.

Reply:
You are right, I was thinking about the same abstraction. If we add anything more into this class, we'll refactor it in that way.